modules/__tests__/Link-test.js

```diff
@@ -557,4 +557,48 @@ describe('A <Link>', () => {
       })
     })
   })
+
+  describe('when the "noHref" prop is specified', function () {
+    class App extends Component {
+      render() {
+        return (
+          <div>
+            <Link to="/hello?the=query#hash" noHref>Link with noHref</Link>
+          </div>
+        )
+      }
+    }
+
+    it('returns an anchor tag without an href', function (done) {
+      render((
+        <Router history={createHistory('/')}>
+          <Route path="/" component={App} />
+        </Router>
+      ), node, function () {
+        const link = node.querySelectorAll('a')[0]
+        expect(link.href).toEqual('')
+        done()
+      })
+    })
+
+    it('should transition correctly on click', done => {
+      const steps = [
+        () => {
+          click(node.querySelector('a'), { button: 0 })
+        },
+        ({ location }) => {
+          expect(location.pathname).toEqual('/hello')
+          expect(location.hash).toEqual('#hash')
+        }
+      ]
+
+      const execNextStep = execSteps(steps, done)
+
+      render((
+        <Router history={createHistory('/hello')} onUpdate={execNextStep}>
+          <Route path="/hello" component={App} />
+        </Router>
+      ), node, execSteps)
```

Comment:
I'm sure because I looked at the [`execStep` code](https://github.com/salemove/react-router/blob/master/modules/__tests__/execSteps.js) and the [documentation for `render`](https://facebook.github.io/react/docs/react-dom.html#render).

All the examples above, except one, use `execNextStep`. The one that uses `execSteps` is broken, but it passes because only the first step will be executed and the second step, which has the assertions, will not be executed and so nothing breaks, but nothing is tested either.